popup_offer/popup_offer.module

```diff
@@ -1,38 +0,0 @@
-<?php
-
-/**
- * Implements hook_preprocess_html().
- */
-function popup_offer_preprocess_html(&$variables)
-{
-  /* @var \Drupal\user\UserDataInterface $userData */
-  $userData = \Drupal::service('user.data');
-  $user = \Drupal::currentUser();
-  $product_builder = \Drupal::service('commerce_product.variation_field_renderer');
-  $builder = \Drupal::service('poppup_offer.special_discount');
-  $p_datas = $userData->get('popup_offer', $user->id(), 'discount_products');
-  if (!empty($p_datas)) {
-    $output = [];
-    $jcnt = 0;
-    foreach ($p_datas['variations'] as $variant) {
-      $output[] = $product_builder->renderFields($variant, 'teaser');
-      $jcnt++;
-    }
-    $icnt = 0;
-    foreach ($p_datas['products'] as $product) {
-      $output[$icnt] += $builder->addToCartForm($product, 'default', TRUE, 'en', $p_datas['discount']);
-      $icnt++;
-    }
-    $variables['page_top']['slidedown_templates'] = [
-        '#theme' => 'modal_page_modal',
-        '#do_not_show_again' => 1,
-        '#title' => 'Each product has ' . ($p_datas['discount'] * 100) . '% discount.',
```

Comment:
t()?

Reply:
yes t() with parameter to substitue the value.